storm-core/src/jvm/org/apache/storm/metrics2/reporters/JmxStormReporter.java

```diff
@@ -67,7 +68,7 @@ public void prepare(MetricRegistry metricsRegistry, Map<String, Object> stormCon
     }
 
     public static String getMetricsJMXDomain(Map reporterConf) {
-        return Utils.getString(reporterConf, JMX_DOMAIN);
+        return Utils.getString(reporterConf.get(Config.STORM_DAEMON_METRICS_REPORTER_PLUGIN_DOMAIN), null);
```

Comment:
JMX_DOMAIN used initially is different than Config.STORM_DAEMON_METRICS_REPORTER_PLUGIN_DOMAIN

I think we should stick to JMX_DOMAIN config key unless @ptgoetz  thinks this is good.

Reply:
I think user should be able to specify a domain so it cannot be a fixed value like JMX_DOMAIN. So the proposed changes looks fine.

Reply:
I think JMX_DOMAIN was intended to be the default value of the domain.

So may be `Utils.getString(reporterConf.get(Config.STORM_DAEMON_METRICS_REPORTER_PLUGIN_DOMAIN), JMX_DOMAIN)`

cc @ptgoetz

Reply:
@arunmahadevan I am not saying we should return that as default value. I am suggesting to use that as the key to look up the conf map.

Reply:
happy to oblige, would like to draw a consensus as in branch 2.x I can't just import Config class, be it due to moving of the class from store-core to storm-client or something else? It's explicitly looking for the key in the Config class?

Reply:
@priyank5485 is correct. JMX_DOMAIN is the key for the lookup from the r Porter config.

Reply:
‘return Utils.getString(reporterConf, JMX_DOMAIN);’

Should be something like 

‘return Utils.getString(reporterConf.get(JMX_DOMAIN), “localhost”);’

(Apologies, not on a computer atm)

Reply:
`Utils.getString(reporterConf.get(JMX_DOMAIN), null)` would keep the intention and avoid NPE.